src/output-json-mqtt.c

```diff
@@ -54,13 +54,17 @@ typedef struct LogMQTTFileCtx_ {
 
 typedef struct LogMQTTLogThread_ {
     LogMQTTFileCtx *mqttlog_ctx;
-    uint32_t        count;
     OutputJsonThreadCtx *ctx;
 } LogMQTTLogThread;
 
-bool JsonMQTTAddMetadata(void *vtx, JsonBuilder *js)
+bool JsonMQTTAddMetadata(void *vtx, JsonBuilder *js, void *ctx)
 {
-    return rs_mqtt_logger_log(vtx, MQTT_DEFAULTS, js);
+    LogMQTTFileCtx *mqtt_ctx = NULL;
+    if (ctx == NULL) {
+        return rs_mqtt_logger_log(vtx, MQTT_DEFAULTS, js);
+    }
+    mqtt_ctx = (LogMQTTFileCtx *)ctx;
```

Comment:
Nice, could we then get rid of src/output-json-mqtt.c file to have it all in rust ?

Reply:
Also ALPROTO_DHCP may be worth a look cf output.c     { ALPROTO_DHCP, NULL }, // TODO missing

Reply:
https://redmine.openinfosecfoundation.org/issues/6091